### DISCUSSION TITLE
Problem on windows installation

### DISCUSSION THREAD

**Discussion by tizfra** [GeekFunkLabs/fluidpatcher#40] [Q&A] [answered]
Hi,
I have a problem on windows installation:
Pyton can't install "wxPython-4.1.1.tar.gz".
Can someone helps me?

**Comment by albedozero** [comment]
Can you paste the output you get from typing in the command
```
python -m pip install oyaml wxpython
```

**Comment by ebrown3k** [comment]
I am having a similar issue. When it tries to install 'wxPython-4.1.1.tar.gz" it fails. I'm not sure how to grab the output from the cli during the install because it closes the window automatically before I can read the full error. I see something about a Legacy Install? I ran --upgrade pip and then tried installing FluidPatcher again and I have the same issue.

**Comment by albedozero** [comment]
Oops - I realize now I put the wrong command in my original reply (now edited). Please type the following into a command window (Windows key + "R" and enter `cmd`) and reply here with the output.
```
python -m pip install oyaml wxpython
```

**Reply by ebrown3k** [reply]
Here's my output.
```
Collecting oyaml
  Using cached oyaml-1.0-py2.py3-none-any.whl (3.0 kB)
Collecting wxpython
  Using cached wxPython-4.1.1.tar.gz (66.0 MB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: pyyaml in c:\users\erik brown\appdata\local\packages\pythonsoftwarefoundation.python.3.10_qbz5n2kfra8p0\localcache\local-packages\python310\site-packages (from oyaml) (6.0)
Requirement already satisfied: pillow in c:\users\erik brown\appdata\local\packages\pythonsoftwarefoundation.python.3.10_qbz5n2kfra8p0\localcache\local-packages\python310\site-packages (from wxpython) (9.1.0)
Requirement already satisfied: six in c:\users\erik brown\appdata\local\packages\pythonsoftwarefoundation.python.3.10_qbz5n2kfra8p0\localcache\local-packages\python310\site-packages (from wxpython) (1.16.0)
Requirement already satisfied: numpy in c:\users\erik brown\appdata\local\packages\pythonsoftwarefoundation.python.3.10_qbz5n2kfra8p0\localcache\local-packages\python310\site-packages (from wxpython) (1.22.3)
Using legacy 'setup.py install' for wxpython, since package 'wheel' is not installed.
Installing collected packages: wxpython, oyaml
  Running setup.py install for wxpython ... error
  error: subprocess-exited-with-error

  × Running setup.py install for wxpython did not run successfully.
  │ exit code: 1
  ╰─> [51 lines of output]
      C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.1264.0_x64__qbz5n2kfra8p0\lib\site-packages\setuptools\dist.py:717: UserWarning: Usage of dash-separated 'license-file' will not be supported in future versions. Please use the underscore name 'license_file' instead
        warnings.warn(
      C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.1264.0_x64__qbz5n2kfra8p0\lib\site-packages\setuptools\dist.py:294: DistDeprecationWarning: use_2to3 is ignored.
        warnings.warn(f"{attr} is ignored.", DistDeprecationWarning)
      running install
      running build
      C:\Users\Erik Brown\AppData\Local\Temp\pip-install-p9kelga5\wxpython_1a0b47699ff74da896ca2d8152b2ba4b\build.py:41: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
        from distutils.dep_util import newer, newer_group
      Will build using: "C:\Users\Erik Brown\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\python.exe"
      3.10.4 (tags/v3.10.4:9d38120, Mar 23 2022, 23:13:41) [MSC v.1929 64 bit (AMD64)]
      Python's architecture is 64bit
      cfg.VERSION: 4.1.1

      Running command: build
      Running command: build_wx
      Command '"C:\Users\Erik Brown\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\python.exe" -c "import setuptools, distutils.msvc9compiler as msvc; mc = msvc.MSVCCompiler(); mc.initialize(); print(mc.cc)"' failed with exit code 1.
      Traceback (most recent call last):

        File "<string>", line 1, in <module>

        File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.1264.0_x64__qbz5n2kfra8p0\lib\distutils\msvc9compiler.py", line 371, in initialize

          vc_env = query_vcvarsall(VERSION, plat_spec)

        File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.1264.0_x64__qbz5n2kfra8p0\lib\site-packages\setuptools\msvc.py", line 140, in msvc9_query_vcvarsall

          return EnvironmentInfo(arch, ver).return_env()

        File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.1264.0_x64__qbz5n2kfra8p0\lib\site-packages\setuptools\msvc.py", line 1740, in return_env

          [self.VCIncludes,

        File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.1264.0_x64__qbz5n2kfra8p0\lib\site-packages\setuptools\msvc.py", line 1282, in VCIncludes

          return [join(self.si.VCInstallDir, 'Include'),

        File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.1264.0_x64__qbz5n2kfra8p0\lib\site-packages\setuptools\msvc.py", line 840, in VCInstallDir

          raise distutils.errors.DistutilsPlatformError(msg)

      distutils.errors.DistutilsPlatformError: Microsoft Visual C++ 14.2 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      Finished command: build_wx (0.282s)
      Finished command: build (0.282s)
      WARNING: Building this way assumes that all generated files have been
      generated already.  If that is not the case then use build.py directly
      to generate the source and perform the build stage.  You can use
      --skip-build with the bdist_* or install commands to avoid this
      message and the wxWidgets and Phoenix build steps in the future.

      "C:\Users\Erik Brown\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\python.exe" -u build.py build
      Command '"C:\Users\Erik Brown\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\python.exe" -u build.py build' failed with exit code 1.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> wxpython

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

**Comment by albedozero** [comment, marked as answer]
Thanks for the info @ebrown3k - that helped me track down the issue. The current version of wxpython doesn't have a pre-compiled binary that is compatible with the latest Python (version 3.10), and most people won't have the MSVC++ compiler to compile it from source (and getting MSVC++ is not trivial either).

Fortunately, wxpython.org provides pre-release compiled versions, and you can tell `pip` to look for these if necessary. If you grab the [latest release](https://github.com/albedozero/fluidpatcher/releases/tag/v0.7.4) of the fluidpatcher Windows installer it will do this automatically, or you can manually install wxpython using the command
```
python -m pip install -U --pre -f https://wxpython.org/Phoenix/snapshot-builds/ wxpython
```

**Reply by ebrown3k** [reply]
Thanks for the quick update. I don't seem to get any errors on install now. However, FluidPatcher won't launch. I tried uninstalling it and reinstalling using the latest version with Python 3.10 compatibility. I have also restarted my computer, just for giggles. It appears to fully install with no errors, but double clicking on the created shortcut doesn't seem to do anything. I don't see any tasks in task manager. I do get an hour glass for about half a second and then nothing. I don't have a ton of time to trouble shoot at the moment, but wanted to get this reply up in case I was missing something totally obvious.

**Reply by albedozero** [reply]
You must have some runtime bug I'm not getting. Somewhat annoyingly, Python just quits silently in GUI mode if it encounters an error, since it doesn't have a console to print the error on. I [modified](https://github.com/albedozero/fluidpatcher/commit/d560486468f9d147706c0924f3bfff931f25833b) fluidpatcher.pyw to print the errors to a dialog box before printing. When you have a moment, would you reinstall [this version](https://github.com/albedozero/fluidpatcher/releases/tag/v0.7.4) and paste a screen-snip of the error? Thanks for your help and time!

**Reply by tizfra** [reply]
Hi everyone,
I've installed last version, but I've this output on installing wxpyton
```
C:\Users\tizza>python -m pip install -U --pre -f https://wxpython.org/Phoenix/snapshot-builds/ wxpython
Looking in links: https://wxpython.org/Phoenix/snapshot-builds/
Collecting wxpython
  Using cached https://wxpython.org/Phoenix/snapshot-builds/wxPython-4.1.2a1.dev5330%2B3e6be81d.tar.gz (70.7 MB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: pillow in c:\users\tizza\appdata\local\programs\python\python310-32\lib\site-packages (from wxpython) (9.1.0)
Requirement already satisfied: six in c:\users\tizza\appdata\local\programs\python\python310-32\lib\site-packages (from wxpython) (1.16.0)
Requirement already satisfied: numpy in c:\users\tizza\appdata\local\programs\python\python310-32\lib\site-packages (from wxpython) (1.22.3)
Using legacy 'setup.py install' for wxpython, since package 'wheel' is not installed.
Installing collected packages: wxpython
  Running setup.py install for wxpython ... error
  error: subprocess-exited-with-error

  × Running setup.py install for wxpython did not run successfully.
  │ exit code: 1
  ╰─> [22 lines of output]
      C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\lib\site-packages\setuptools\dist.py:717: UserWarning: Usage of dash-separated 'license-file' will not be supported in future versions. Please use the underscore name 'license_file' instead
        warnings.warn(
      C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\lib\site-packages\setuptools\dist.py:294: DistDeprecationWarning: use_2to3 is ignored.
        warnings.warn(f"{attr} is ignored.", DistDeprecationWarning)
      running install
      running build
      C:\Users\tizza\AppData\Local\Temp\pip-install-dpvppr1k\wxpython_4ddc4254302d4e1fbee52db4616d9ce1\build.py:42: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
        from distutils.dep_util import newer, newer_group
      Traceback (most recent call last):
        File "C:\Users\tizza\AppData\Local\Temp\pip-install-dpvppr1k\wxpython_4ddc4254302d4e1fbee52db4616d9ce1\build.py", line 49, in <module>
          from buildtools.wxpysip import sip_runner
        File "C:\Users\tizza\AppData\Local\Temp\pip-install-dpvppr1k\wxpython_4ddc4254302d4e1fbee52db4616d9ce1\buildtools\wxpysip.py", line 20, in <module>
          from sipbuild.code_generator import (set_globals, parse, generateCode,
      ModuleNotFoundError: No module named 'sipbuild'
      WARNING: Building this way assumes that all generated files have been
      generated already.  If that is not the case then use build.py directly
      to generate the source and perform the build stage.  You can use
      --skip-build with the bdist_* or install commands to avoid this
      message and the wxWidgets and Phoenix build steps in the future.

      "C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\python.exe" -u build.py build
      Command '"C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\python.exe" -u build.py build' failed with exit code 1.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> wxpython

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
WARNING: You are using pip version 22.0.4; however, version 22.1 is available.
You should consider upgrading via the 'C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\python.exe -m pip install --upgrade pip' command.
```

This is the same error I have at the beginning.

**Reply by ebrown3k** [reply]
I installed the new version. I had to uninstall the old one first because it wasn't giving me the new file in the install directory. I still had the same experience with the desktop shortcut that noting appeared to happen. So I tried PowerShell and running it from the cli. Here is what it gave me: 

```
PS C:\Program Files\FluidPatcher> python fluidpatcher.pyw
Error in sys.excepthook:
Traceback (most recent call last):
  File "C:\Program Files\FluidPatcher\fluidpatcher.pyw", line 28, in gui_excepthook
    wx.MessageBox(''.join(s), "Error", wx.OK|wx.ICON_ERROR)
wx._core.PyNoAppError: The wx.App object must be created first!

Original exception was:
Traceback (most recent call last):
  File "C:\Program Files\FluidPatcher\fluidpatcher.pyw", line 508, in <module>
    pxr = patcher.Patcher(cfgfile)
  File "C:\Program Files\FluidPatcher\patcher\__init__.py", line 39, in __init__
    self.read_config()
  File "C:\Program Files\FluidPatcher\patcher\__init__.py", line 84, in read_config
    raw = self.cfgfile.read_text()
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.1264.0_x64__qbz5n2kfra8p0\lib\pathlib.py", line 1132, in read_text
    with self.open(mode='r', encoding=encoding, errors=errors) as f:
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.1264.0_x64__qbz5n2kfra8p0\lib\pathlib.py", line 1117, in open
    return self._accessor.open(self, mode, buffering, encoding, errors,
FileNotFoundError: [Errno 2] No such file or directory: 'fluidpatcherconf.yaml'
PS C:\Program Files\FluidPatcher>
```

**Reply by albedozero** [reply]
@tizfra the output that states `package 'wheel' is not installed` is odd - maybe your windows %PATH% is pointing to another version of python you have installed? I'm realizing I should be using the Windows launcher `py` in my commands rather than `python`, so that the correct python version is used. At any rate, try
```
py -m pip install wheel
py -m pip install -U --pre -f https://wxpython.org/Phoenix/snapshot-builds/ wxpython
```

**Reply by albedozero** [reply]
@ebrown3k good idea trying to run the program from a shell, but you need the location of the config file on the command line, and I store this in the user's "Documents" directory rather than "Program Files" so it can be user-editable. Try
```
py fluidpatcher.pyw %HOMEPATH%\Documents\FluidPatcher\fluidpatcherconf.yaml
```

**Reply by ebrown3k** [reply]
Ah, that makes sense. Running from cli with the full path to the config it actually launches with only a couple lines of message (see below). I don't get any sound. I'm not sure if that's what the warnings are referring to or if I just don't have the settings for my machine correct yet. I can see midi notes on/off in the midi monitor when I bang on my midi controller and I can navigate patches with the buttons in the GUI. I still can't launch from the shortcut or double clicking the fluidpatcher.pyw file in windows explorer.

```
fluidsynth: warning: FluidSynth has not been compiled with LADSPA support
fluidsynth: warning: Sample 'SineWave': ROM sample ignored
```

**Reply by albedozero** [reply]
The warnings are normal for Windows and shouldn't stop sound from working - the first means that external effects plugins won't work, and the second refers to an old ROM sample left in VintageDreamsWaves-v2.sf2 that isn't needed for anything. The fact that no error is showing up, but you still have no sound and can't run from the shortcut makes me think something's up with your config file. Can you paste the text from your Tools -> Settings... dialog in FluidPatcher? Also, if you reinstall using my latest, latest update to the [v0.7.4 release](https://github.com/albedozero/fluidpatcher/releases/tag/v0.7.4) the `wx.App object must be created first` bug from above is fixed, and you should get some sort of error output in the dialog if there is any. Make sure to uninstall first per @tizfra's comment above.

**Reply by ebrown3k** [reply]
I am running the 0.7.4 version acording the the About dialog box. Here is my config which I have not edited.


```
soundfontdir: C:\Users\Erik Brown\Documents\FluidPatcher\sf2
bankdir: C:\Users\Erik Brown\Documents\FluidPatcher\banks
mfilesdir: C:\Users\Erik Brown\Documents\FluidPatcher\midi
currentbank: bank1.yaml
fluidsettings:
  midi.autoconnect: 1
  player.reset-synth: 0
  synth.ladspa.active: 1
fluidsettings-windows-lowlatency:
  audio.driver: wasapi
  audio.wasapi.exclusive-mode: 1
  audio.periods: 16
  audio.period-size: 144
  midi.autoconnect: 1
  player.reset-synth: 0
fluidsettings-linux-jack:
  audio.driver: jack
  audio.jack.autoconnect: 1
  audio.jack.multi: 1
  midi.autoconnect: 1
  player.reset-synth: 0
  synth.audio-channels: 16
  synth.audio-groups: 16
  synth.ladspa.active: 1
```

And thank you for all your help. I hope all of this is helping you make a better product and not just me randomly flailing away on something that is an issue on my end.

**Reply by albedozero** [reply]
Definitely - your feedback is much appreciated and helpful. The config file does look fine. So it seems like there are two issues: no sound, and launching the program from the shortcut crashes for some unknown reason.

- **no sound**: FluidPatcher should be using Windows DirectSound by default, but you can specify this by adding a line `audio.driver: dsound` in the `fluidsettings:` section of your config file (BTW the `windows-lowlatency` and `linux-jack` sections don't do anything, they're just there as examples). FluidPatcher will use the default playback device you have selected in the Windows "Sound" control panel item - if you have multiple sound output devices make sure the one you want is set as default. You can also set the device FluidPatcher uses explicity with the `audio.dsound.device` fluidsetting, but I'm not sure how Windows names the devices. If that doesn't work, you could try some of the other Windows drivers listed [here](https://github.com/FluidSynth/fluidsynth/wiki/Audio-Drivers).
- **no launch from shortcut**: I'm not sure why this would work from the command line but not the shortcut. If you have a chance to reinstall using the updated release maybe we can get some feedback.

**Reply by tizfra** [reply]
Hi, thanks for the help,
I've removed all Python version in my PC, then I've installed Fuidpatcher and I got also an error.
So I tried run py `-m pip install -U --pre -f https://wxpython.org/Phoenix/snapshot-builds/ wxpython`
and I got this:
```
Looking in links: https://wxpython.org/Phoenix/snapshot-builds/
Collecting wxpython
  Using cached https://wxpython.org/Phoenix/snapshot-builds/wxPython-4.1.2a1.dev5330%2B3e6be81d.tar.gz (70.7 MB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: pillow in c:\users\tizza\appdata\local\programs\python\python310-32\lib\site-packages (from wxpython) (9.1.1)
Requirement already satisfied: six in c:\users\tizza\appdata\local\programs\python\python310-32\lib\site-packages (from wxpython) (1.16.0)
Requirement already satisfied: numpy in c:\users\tizza\appdata\local\programs\python\python310-32\lib\site-packages (from wxpython) (1.22.3)
Building wheels for collected packages: wxpython
  Building wheel for wxpython (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [22 lines of output]
      C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\lib\site-packages\setuptools\dist.py:772: UserWarning: Usage of dash-separated 'license-file' will not be supported in future versions. Please use the underscore name 'license_file' instead
        warnings.warn(
      C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\lib\site-packages\setuptools\config\setupcfg.py:459: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
        warnings.warn(msg, warning_class)
      C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\lib\site-packages\setuptools\dist.py:318: DistDeprecationWarning: use_2to3 is ignored.
        warnings.warn(f"{attr} is ignored.", DistDeprecationWarning)
      running bdist_wheel
      running build
      Traceback (most recent call last):
        File "C:\Users\tizza\AppData\Local\Temp\pip-install-vh6b6pcg\wxpython_dd342fc046e144eeb635b7e372544447\build.py", line 49, in <module>
          from buildtools.wxpysip import sip_runner
        File "C:\Users\tizza\AppData\Local\Temp\pip-install-vh6b6pcg\wxpython_dd342fc046e144eeb635b7e372544447\buildtools\wxpysip.py", line 20, in <module>
          from sipbuild.code_generator import (set_globals, parse, generateCode,
      ModuleNotFoundError: No module named 'sipbuild'
      WARNING: Building this way assumes that all generated files have been
      generated already.  If that is not the case then use build.py directly
      to generate the source and perform the build stage.  You can use
      --skip-build with the bdist_* or install commands to avoid this
      message and the wxWidgets and Phoenix build steps in the future.

      "C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\python.exe" -u build.py build
      Command '"C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\python.exe" -u build.py build' failed with exit code 1.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for wxpython
  Running setup.py clean for wxpython
Failed to build wxpython
Installing collected packages: wxpython
  Running setup.py install for wxpython ... error
  error: subprocess-exited-with-error

  × Running setup.py install for wxpython did not run successfully.
  │ exit code: 1
  ╰─> [24 lines of output]
      C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\lib\site-packages\setuptools\dist.py:772: UserWarning: Usage of dash-separated 'license-file' will not be supported in future versions. Please use the underscore name 'license_file' instead
        warnings.warn(
      C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\lib\site-packages\setuptools\config\setupcfg.py:459: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
        warnings.warn(msg, warning_class)
      C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\lib\site-packages\setuptools\dist.py:318: DistDeprecationWarning: use_2to3 is ignored.
        warnings.warn(f"{attr} is ignored.", DistDeprecationWarning)
      running install
      C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      Traceback (most recent call last):
        File "C:\Users\tizza\AppData\Local\Temp\pip-install-vh6b6pcg\wxpython_dd342fc046e144eeb635b7e372544447\build.py", line 49, in <module>
          from buildtools.wxpysip import sip_runner
        File "C:\Users\tizza\AppData\Local\Temp\pip-install-vh6b6pcg\wxpython_dd342fc046e144eeb635b7e372544447\buildtools\wxpysip.py", line 20, in <module>
          from sipbuild.code_generator import (set_globals, parse, generateCode,
      ModuleNotFoundError: No module named 'sipbuild'
      WARNING: Building this way assumes that all generated files have been
      generated already.  If that is not the case then use build.py directly
      to generate the source and perform the build stage.  You can use
      --skip-build with the bdist_* or install commands to avoid this
      message and the wxWidgets and Phoenix build steps in the future.

      "C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\python.exe" -u build.py build
      Command '"C:\Users\tizza\AppData\Local\Programs\Python\Python310-32\python.exe" -u build.py build' failed with exit code 1.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> wxpython

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```
It looks like an error during compiling.
My system is 64bit, but I try installing 32bit version of Fluidpatcher, can this cause this error?
Thanks in advance.

**Reply by tizfra** [reply]
Ok, i delete all 32bit software (python and fluidpatcher) and installed 64bit, and now it starts, but don't read any midi message from my keyboard, but on MIDI-OX I read midi incoming.

**Reply by albedozero** [reply]
Aha - there are _no_ versions of wxpython available for 32-bit Python 3.10. I've modified the 32-bit installer to use Python 3.9 instead. I've also modified both installers to use the Windows launcher `py` to be more specific about which Python version they should use - this may fix @ebrown3k 's problem with the shortcut not working.

For the no sound problem - is your keyboard sending notes on MIDI channel 1? Just an FYI - on Windows only one program can use a given MIDI device at one time, so if you have MIDI-OX open, FluidPatcher will not receive MIDI signals from your keyboard.

**Reply by tizfra** [reply]
I opened fluidpatches then i connected my keyboard and i tried, no sounds coming out so I opened midi monitor and i saw that no midi messages coming.
I closed fluidpatches and start midihox and i found midi messages coming to PC.
Yes, keyboard send on channel 1 and works well on squishbox

**Reply by albedozero** [reply]
Hmm - @tizfra and @ebrown3k let me see if I can get some more information about why FluidSynth can't seem to connect to your MIDI controller(s) and/or produce audio. If you unzip [getinfo.zip](https://github.com/albedozero/fluidpatcher/files/8712719/getinfo.zip) into your "%PROGRAMFILES%"\FluidPatcher directory and run _getinfo.bat_, it will print some information about the audio/midi drivers and devices available to FluidSynth on your system. If you don't want to paste that info here feel free to email it to the address in my github profile. Thanks for your help!

**Reply by ebrown3k** [reply]
@tizfra I found I had to have my midi controller plugged in when I launched FluidPatcher to see midi notes in midi monitor. Maybe try that. @albedozero I will have a chance to try installing the newest newest latest version later today and let you know how it goes.

**Comment by ebrown3k** [comment]
So I uninstalled and then reinstalled with the latest version. When I double clicked the desktop icon I get a missing shortcut error:
![missing shortcut](https://user-images.githubusercontent.com/15970822/169139369-96b186a3-3e8c-4d53-96ae-c9a7e6e7f2ab.PNG)
I browsed and connected it to Program Files/FluidPatcher/fluidpatcher.pyw and I get the below error:
![runtime](https://user-images.githubusercontent.com/15970822/169139372-688ccdbd-ef16-4d8c-b994-44dde45c28a3.PNG)
I also ran the getinfo batch and see it is missing soundfonts. So maybe I just missed the instructions telling me I needed to create a soundfont folder and put a sf2 file in it. But that still doesn't explain why the desktop icon won't work, or now the new errors I'm seeing.

```
audio.driver:
Type:          string
Value:         dsound
Default value:
Options:       dsound, file, wasapi, waveout
Real-time:     no
audio.dsound.device:
Type:          string
Value:         default
Default value: default
Options:       E320-B0 (NVIDIA High Definition Audio), Primary Sound Driver, Speakers (Realtek(R) Audio), Speakers (Sound Blaster Play! 3), default
Real-time:     no
midi.driver:
Type:          string
Value:         winmidi
Default value:
Options:       winmidi
Real-time:     no
midi.winmidi.device:
Type:          string
Value:         default
Default value: default
Options:       0:Q49 MKII, 1:MIDIIN2 (Q49 MKII), default
Real-time:     no
fluidsynth: error: fluid_sfloader_load(): Failed to open 'C:\soundfonts\default.sf2': File does not exist.
fluidsynth: error: Unable to open file 'C:\soundfonts\default.sf2'
fluidsynth: error: ipatch_file_open() failed with error: 'No such file or directory'
fluidsynth: error: Failed to load SoundFont "C:\soundfonts\default.sf2"
audio.driver:
Type:          string
Value:         dsound
Default value:
Options:       dsound, file, wasapi, waveout
Real-time:     no
audio.dsound.device:
Type:          string
Value:         default
Default value: default
Options:       E320-B0 (NVIDIA High Definition Audio), Primary Sound Driver, Speakers (Realtek(R) Audio), Speakers (Sound Blaster Play! 3), default
Real-time:     no
midi.driver:
Type:          string
Value:         winmidi
Default value:
Options:       winmidi
Real-time:     no
midi.winmidi.device:
Type:          string
Value:         default
Default value: default
Options:       0:Q49 MKII, 1:MIDIIN2 (Q49 MKII), default
Real-time:     no
```

**Comment by albedozero** [comment]
`py` and `pyw` are the Python launchers for console and GUI, respectively. They should live in C:\Windows - that's what you should point the shortcut to rather than fluidpatcher.pyw. They get installed when you install any flavor of python, and they can be used to specify which version your'e trying to run. It seems like your system doesn't have it installed, maybe b/c you installed Python from the Windows app store, plus the installer wasn't properly detecting whether python was installed (argh!). Try the latest latest latest version of the installer (sorry 🙄).

The errors about the default soundfont not being found when _getinfo.bat_ is run are nothing to worry about - it's because the batch file is running FluidSynth standalone - FluidPatcher loads the proper soundfonts based on what is in the current bank file. Try specifying `audio.dsound.device` and `midi.winmidi.device` with one of the devices listed in the _Options_ items in your output above - maybe the defaults aren't working for you for some reason.

**Comment by ebrown3k** [comment]
Success! I think there were a number of issues here. Somehow my Python was not working properly. I know I had installed it myself from the Python repository because I have been attempting to learn Python and had done some exercises out of the book, but that's been a few months ago so maybe something happened to it. This last installer finally detected that I didn't have Python and had me install it. The desktop shortcut works. Why I was able to get it to launch from the cli previously if my Python was corrupted or missing, I don't know.

The next part was 100% user error. I thought I was transmitting on MIDI channel 1, but apparently I was not. So that was my no sound issue. The MIDI monitor saw notes, they just weren't on the right channel. 

Now the next step will be to get my SquishBox running. I built it from the kit last weekend but the power supply I thought I had didn't fit so I had to wait to get a new one. I have that now and with FluidPatcher working on Windows now, hopefully this next bit goes smoother.

Thank you, @albedozero, for all the help!
